website/docs/components/form/primitives/partials/guidelines/guidelines.md

```diff
@@ -9,6 +9,51 @@
 
 - Use helper text to give the user extra details about the data you’re asking them to input, e.g., formatting requirements such as MM-DD-YYYY.
 
+### Form::CharacterCount
+
+Use a character count to communicate the current length of the value in an input and whether it meets or exceeds the length requirements passed to the component. The component accepts multiple arguments to set length requirements and exposes several computed values to support custom messages. Visit the [code](/components/form/primitives?tab=code#formcharactercount-1) documentation for more details. <!-- TODO: ensure that this link is correct -->
+
+#### Default messages
+
+Depending on which property (or properties) are passed to the component, a different default message will be displayed by the component to communicate the relationship between the current length of the input value (`currentLength`) and the maximum length (`maxLength`), minimum length (`minLength`), or both.
+
+The default messages provide a consistent messaging pattern for the component by clearly communicating length requirements to the user while displaying their progress towards meeting the requirements.
+
+!!! Info
+
+For representative consistenty, the Figma component mirrors the default messages that are rendered in the Ember component and are labelled as such; `default` (or `currentLength`), `maxLength`, `minLength`, and `custom`. However, if anything apart from the numerical value (e.g., "{numerical value} characters is allowed") in the non-custom variants is edited, this will require a custom implementation on the engineering side to match.
```

Comment:
The reason I included this is maybe a designer might want to create a prototype or something that is more directly representative, in which they may need to change the number.

Reply:
Changing the number wouldn't require a custom implementation, though, right?

Reply:
Right, changing the number wouldn't, but it's a bit hard to message since the defaults in the Ember component handle the increment and count method out of the box. Let me think on this a little bit to see if there's a line that can be drawn

Reply:
Added a bit more clarity in https://github.com/hashicorp/design-system/pull/1799/commits/beab5d75359648411a042f71e4d8817f06b5c9df, what do you think?